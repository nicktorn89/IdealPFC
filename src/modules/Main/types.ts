export interface MainState {
  weight: number;
  height: number;
  age: number;
  desiredWeight?: number;

  [key: string]: number | undefined;
}

export interface MainProps {
}

export interface Nutrients {
  protein: number;
  fats: number;
  carbohydrates: number;
}
