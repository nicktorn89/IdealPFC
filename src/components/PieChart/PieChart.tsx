import React from 'react';
import { Section } from 'react-bulma-components/full';
import { ChartsProps } from './types';
import { PieChart, Pie, Cell } from 'recharts';

const PieChartComponent: React.FC<ChartsProps> = ({ data, size = 50 }) =>
  <Section>
    <PieChart width={330} height={330}>
      <Pie
        data={data}
        dataKey='value'
        nameKey='name'
        cx='50%'
        cy='50%'
        outerRadius={size}
      >
        {data.map((entry, index) => <Cell key={index} fill={entry.color} />)}
      </Pie>
    </PieChart>
  </Section>;

PieChartComponent.displayName = 'PieChart';

export default PieChartComponent;
