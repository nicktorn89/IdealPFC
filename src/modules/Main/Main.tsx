import React from 'react';
import { MainState, MainProps } from './types';
import FormComponent from '../../components/Form';
import HeaderComponent from 'src/components/Header';
import ChartPFC from 'src/components/ChartPFC';
import { PieChartData } from 'src/components/ChartPFC/types';
import LineChart from 'src/components/LineChart';
import { LineChartData } from 'src/components/LineChart/types';

const dataMock: PieChartData[] = [
  {
    name: 'Белки',
    value: 200,
    color: '#FFFF4C',
  },
  {
    name: 'Жиры',
    value: 150,
    color: '#FF4C4C',
  },
  {
    name: 'Углеводы',
    value: 650,
    color: '#4C4CFF',
  },
];

const anotherMock: LineChartData[] = [
  {
    name: '1 неделя',
    calories: 2500,
  },
  {
    name: '2 неделя',
    calories: 2300,
  },
  {
    name: '3 неделя',
    calories: 2100,
  },
  {
    name: '4 неделя',
    calories: 1900,
  },
  {
    name: '5 неделя',
    calories: 1700,
  },
];

export class Main extends React.Component<MainProps, MainState> {
  public readonly state = {
    weight: 0,
    height: 0,
    age: 0,
  };

  public handleFieldChange = ({ target }: React.MouseEvent<HTMLInputElement>) => {
    const { name, value } = target as HTMLInputElement;

    if (isNaN(+value)) return;

    this.setState({ [name]: +value });
  }

  public render = () => {
    return (
      <React.Fragment>
        <HeaderComponent />
        <FormComponent
          values={this.state}
          onChangeField={this.handleFieldChange}
        />
        <ChartPFC data={dataMock} />
        <LineChart data={anotherMock} />
      </React.Fragment>
    );
  }
}

export default Main;
