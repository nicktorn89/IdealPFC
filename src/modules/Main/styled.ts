import styled from 'styled-components';
import { media } from 'src/theme/mixin';
