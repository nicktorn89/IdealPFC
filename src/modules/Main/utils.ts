import { Nutrients } from './types';

export const roundToDecimal = (value: number) => Math.round(value * 10) / 10;

export const countNutrients = (height: number, weight: number, hasSportLoad?: boolean): Nutrients => 
  ({
    protein: roundToDecimal(weight * (hasSportLoad ? 2.5 : 1.5)),
    fats: roundToDecimal(weight * (hasSportLoad ? 1.5 : 0.8)),
    carbohydrates: roundToDecimal(weight * (hasSportLoad ? 4 : 2)),
  });
