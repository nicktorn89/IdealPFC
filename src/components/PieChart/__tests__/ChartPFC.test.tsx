import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { default as Chart } from '../PieChart';
import { dataMock } from 'src/modules/Main/Main';

describe('<PieChart />', () => {
  const PieChart = shallow(<Chart data={dataMock} />);
  const PieChartToJson = shallowToJson(PieChart);

  it('should be equal to snapshot', () => {
    expect(PieChartToJson).toMatchSnapshot();
  });
});
