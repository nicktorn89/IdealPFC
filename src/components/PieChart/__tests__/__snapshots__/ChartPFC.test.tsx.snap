// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChartPFC /> should be equal to snapshot 1`] = `
<c
  className=""
  renderAs="section"
  size={null}
  style={Object {}}
>
  <PieChart
    barCategoryGap="10%"
    barGap={4}
    cx="50%"
    cy="50%"
    endAngle={360}
    height={330}
    innerRadius={0}
    layout="centric"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 5,
      }
    }
    outerRadius="80%"
    reverseStackOrder={false}
    stackOffset="none"
    startAngle={0}
    width={330}
  >
    <Pie
      animationBegin={400}
      animationDuration={1500}
      animationEasing="ease"
      blendStroke={false}
      cx="50%"
      cy="50%"
      data={
        Array [
          Object {
            "color": "#FFFF4C",
            "name": "Белки",
            "value": 200,
          },
          Object {
            "color": "#FF4C4C",
            "name": "Жиры",
            "value": 150,
          },
          Object {
            "color": "#4C4CFF",
            "name": "Углеводы",
            "value": 650,
          },
        ]
      }
      dataKey="value"
      endAngle={360}
      fill="#808080"
      hide={false}
      innerRadius={0}
      isAnimationActive={true}
      labelLine={true}
      legendType="rect"
      minAngle={0}
      nameKey="name"
      outerRadius={50}
      paddingAngle={0}
      startAngle={0}
      stroke="#fff"
    >
      <Cell
        fill="#FFFF4C"
        key="0"
      />
      <Cell
        fill="#FF4C4C"
        key="1"
      />
      <Cell
        fill="#4C4CFF"
        key="2"
      />
    </Pie>
  </PieChart>
</c>
`;
